File: 2017-multi-jpa-java/app/src/main/java/sample/member/app/security/SimpleUserDetailsService.java
package sample.member.app.security;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import sample.member.domain.model.UserEntity;
import sample.member.domain.repository.UserRepository;

import java.util.ArrayList;

public class SimpleUserDetailsService implements UserDetailsService {

	private UserRepository userRepository;

	public SimpleUserDetailsService(UserRepository userRepository) {
		this.userRepository = userRepository;
	}

	@Override
	public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
		UserEntity userEntity = userRepository.findByUsername(username).orElseThrow(() -> new UsernameNotFoundException("사용자가 없습니다."));

		return new SimpleUser(
				userEntity.getUsername(),
				new ArrayList<GrantedAuthority>() {{
					new SimpleGrantedAuthority("ROLE_USER");
				}},
				userEntity.getPassword()
		);
	}
}

File: rabbitmq-jpa-kotlin/app/src/main/code/sample/app/security/SimpleUserDetailsService.kt
package sample.app.security

import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import sample.domain.repository.UserRepository

class SimpleUserDetailsService(private val userRepository: UserRepository) : UserDetailsService {

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails =
        userRepository.findByUsername(username).orElseThrow { UsernameNotFoundException("사용자가 없습니다") }
            .let {
                SimpleUser(
                    it.username,
                    listOf(SimpleGrantedAuthority("ROLE_USER")),
                    it.password
                )
            }
}
